apps/dashboard/src/app/knowledge-box/knowledge-box-process/training-history/training-history.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Component } from '@angular/core';
+import { SDKService } from '@flaps/core';
+import { concatMap, filter, map, scan, tap, take, BehaviorSubject } from 'rxjs';
+import { TrainingExecution, TrainingExecutionStatus } from '@nuclia/core';
+import { intervalToDuration } from 'date-fns';
+
+@Component({
+  selector: 'app-training-history',
+  templateUrl: './training-history.component.html',
+  styleUrls: ['./training-history.component.scss'],
+})
+export class TrainingHistoryComponent {
+  status = TrainingExecutionStatus;
+  isLastPage = false;
+  currentPage = new BehaviorSubject<number>(0);
+  executions = this.currentPage.pipe(
+    concatMap(() =>
+      this.sdk.currentKb.pipe(
+        filter((kb) => !!kb),
+        take(1),
+      ),
+    ),
+    concatMap((kb) => kb.training.getExecutions(this.currentPage.getValue())),
+    tap((data) => (this.isLastPage = data.pagination.last)),
+    map((data) =>
+      data.items.map((item) => ({
+        ...item,
+        duration: this.getDuration(new Date(item.start), new Date(item.end)),
+      })),
+    ),
+    scan((acc, current) => acc.concat(current), [] as (TrainingExecution | { duration: string })[]),
+  );
+
+  constructor(private sdk: SDKService) {}
+
+  getDuration(start: Date, end: Date) {
+    const duration = intervalToDuration({ start, end });
+    return (
+      `${duration.hours?.toString().padStart(2, '0')}:` +
+      `${duration.minutes?.toString().padStart(2, '0')}:` +
+      `${duration.seconds?.toString().padStart(2, '0')}`
+    );
```

Comment:
I'm not sure what you're doing here, but playing with dates and time is usually better to do through the [date-fns](https://date-fns.org). We're using it a lot in Pastanaga (for datetime pipes and calendar component), it's well documented and has plenty of easy to use functions.

Reply:
fixed, I found a function in `date-fns` to format time durations.